datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -342,15 +342,15 @@ impl ExternalSorter {
                 streams.push(stream);
             }
 
-            streaming_merge(
+            streaming_merge(StreamingMergeConfig {
                 streams,
-                Arc::clone(&self.schema),
-                &self.expr,
-                self.metrics.baseline.clone(),
-                self.batch_size,
-                self.fetch,
-                self.reservation.new_empty(),
-            )
+                schema: Arc::clone(&self.schema),
+                expressions: &self.expr,
+                metrics: self.metrics.baseline.clone(),
+                batch_size: self.batch_size,
+                fetch: self.fetch,
+                reservation: self.reservation.new_empty(),
+            })
```

Comment:
Another potential API to consider might be a builder style

```suggestion
            StreamingMergeBuilder::new()
              .with_streams(streams)
              .with_schema(Arc::clone(&self.schema))
              .with_expressions(&self.expr)
              .with_metrics(self.metrics.baseline.clone())
              .with_batch_size(self.batch_size)
              .with_fetch(self.fetch)
              .with_reservation(self.reservation.new_empty())
              .build()?
```

Reply:
👍 for the builder

Reply:
Thank you for the review and suggestion! I've implemented the builder pattern like you suggested.